doc/modules/linear_model.rst

```diff
@@ -61,6 +63,21 @@ example, when data are collected without an experimental design.
 
    * :ref:`sphx_glr_auto_examples_linear_model_plot_ols.py`
 
+.. _non_negative_least_square:
+
+Non-Negative Least Squares
+--------------------------
+
+When the coefficients represent some physical or naturally non-negative
+quantities (e.g., frequency counts or prices of goods), it is possible
+to constrain all the coefficients to be non-negative.
```

Comment:
One may want positive coeffs even when they don't represent some physical quantity so I think the structure should be reversed, e.g.:

> It is possible to constrain all the coefficients to be non-negative, which may be useful when they represent some physical or naturally non-negative
quantities (e.g., frequency counts or prices of goods).